subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/extensions/AbstractMultiTestInterceptor.java

```diff
@@ -107,6 +98,14 @@ void interceptIteration(IMethodInvocation invocation, Execution currentExecution
         }
     }
 
+    private static void resetInterceptors(IMethodInvocation invocation) throws NoSuchFieldException, IllegalAccessException {
+        Iterator<IMethodInterceptor> iterator = invocation.getMethod().getInterceptors().iterator();
+        iterator.next(); // This interceptor should be the first - skip it
```

Comment:
Would it make sense to fail with a reasonable error message if it's not?

Reply:
And/or maybe check that `invocation instanceof MethodInvocation` to have compile time safety for this shortcut